File: app/src/main/java/com/giaquino/android/cropped/data/repository/SharedRepositoryImpl.java
package com.giaquino.android.cropped.data.repository;

import javax.inject.Inject;

public class SharedRepositoryImpl implements SharedRepository {

    private boolean isLoggedIn;
    private String accessToken;

    @Inject
    public SharedRepositoryImpl() {
    }

    @Override
    public boolean isLoggedIn() {
        return isLoggedIn;
    }

    @Override
    public void setLoggedIn(boolean isLoggedIn) {
        this.isLoggedIn = isLoggedIn;
    }

    @Override
    public String getAccessToken() {
        return accessToken;
    }

    @Override
    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }
}

File: app/src/main/java/com/giaquino/android/cropped/data/repository/SharedRepositoryImpl.kt
package com.giaquino.android.cropped.data.repository

import android.app.Application
import androidx.datastore.preferences.core.edit
import androidx.datastore.preferences.core.stringPreferencesKey
import com.giaquino.android.cropped.accountDataStore
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.flow.map
import kotlinx.coroutines.runBlocking
import javax.inject.Inject

class SharedRepositoryImpl @Inject constructor(private val application : Application) : SharedRepository {

    override fun getUnsplashToken(): String? = runBlocking {
        application.accountDataStore.data.map {
            it[UNSPLASH_TOKEN]
        }.first()
    }

    override fun setUnsplashToken(token: String) {
        runBlocking {
            application.accountDataStore.edit {
                it[UNSPLASH_TOKEN] = token
            }
        }
    }

    companion object {
        val UNSPLASH_TOKEN = stringPreferencesKey("unsplash_token")
    }
}